src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,821 @@
+                localIndex q=0;
+
+                real64 invJ[3][3]={{0}};
+                FE_TYPE::invJacobianTransformation( q, xLocal, invJ );
+
+                real64 coordsRef[3]={0};
+                for( localIndex i=0; i<3; ++i )
+                {
+                  coordsRef[i] = coords[i] - xLocal[q][i];
+                }
+
+                for( localIndex i=0; i<3; ++i )
+                {
+                  /// Init at (-1,-1,-1) as the origin of the referential elem
+                  coordsOnRefElem[i] =-1.0;
+                  for( localIndex j=0; j<3; ++j )
+                  {
+                    coordsOnRefElem[i] += invJ[i][j]*coordsRef[j];
+                  }
+                }
+
+                real64 Ntest[8];
+                finiteElement::LagrangeBasis1::TensorProduct3D::value( coordsOnRefElem, Ntest );
+
+                for( localIndex a=0; a< numNodesPerElem; ++a )
+                {
+                  receiverNodeIds[ircv][a] = elemsToNodes[k][a];
+                  receiverConstants[ircv][a] = Ntest[a];
+                }
+              }
+            }
+          } // End loop over reciever
+
+        } // End loop over elements
+      } );
+    } );
+  } );
+}
+
+
+void AcousticWaveEquationSEM::addSourceToRightHandSide( real64 const & time, arrayView1d< real64 > const rhs )
+{
+  arrayView2d< localIndex const > const sourceNodeIds = m_sourceNodeIds.toViewConst();
+  arrayView2d< real64 const > const sourceConstants   = m_sourceConstants.toViewConst();
+  arrayView1d< localIndex const > const sourceIsLocal = m_sourceIsLocal.toViewConst();
+
+  real64 const fi = evaluateRickerOrder2( time, this->m_timeSourceFrequency );
+
+  for( localIndex isrc = 0; isrc < sourceConstants.size( 0 ); ++isrc )
+  {
+    if( sourceIsLocal[isrc] == 1 )
+    {
+      for( localIndex inode = 0; inode < sourceConstants.size( 1 ); ++inode )
+      {
+        rhs[sourceNodeIds[isrc][inode]] = sourceConstants[isrc][inode] * fi;
+      }
+    }
+  }
+}
+
+
+void AcousticWaveEquationSEM::computeSismoTrace( localIndex const isismo, arrayView1d< real64 > const pressure_np1 )
+{
+  arrayView2d< localIndex const > const receiverNodeIds = m_receiverNodeIds.toViewConst();
+  arrayView2d< real64 const > const receiverConstants   = m_receiverConstants.toViewConst();
+  arrayView1d< localIndex const > const receiverIsLocal = m_receiverIsLocal.toViewConst();
+
+  arrayView1d< real64 > const p_rcvs   = m_pressureNp1AtReceivers.toView();
+
+
+  char filename[50];
+
+  for( localIndex ircv = 0; ircv < receiverConstants.size( 0 ); ++ircv )
+  {
+    if( receiverIsLocal[ircv] == 1 )
+    {
+      p_rcvs[ircv] = 0.0;
+      for( localIndex inode = 0; inode < receiverConstants.size( 1 ); ++inode )
+      {
+        p_rcvs[ircv] += pressure_np1[receiverNodeIds[ircv][inode]]*receiverConstants[ircv][inode];
+      }
+
+      sprintf( filename, "sismoTraceReceiver%0ld.txt", ircv );
+      this->saveSismo( isismo, p_rcvs[ircv], filename );
+    }
+  }
+}
```

Comment:
I think you may want to comment out this code when you merge into `develop`. This function is very useful for you before it does something that cannot be done with `TimeHistory` yet, but it is going to create a lot of text files in the `integratedTests` repository (and these may not always be cleaned up by the ats script that we use, see https://github.com/GEOSX/integratedTests/issues/120). So I would suggest to comment out this function.